Src/Coravel/EventServiceRegistration.cs

```diff
@@ -19,10 +20,85 @@ public static IServiceCollection AddEvents(this IServiceCollection services)
         public static IEventRegistration ConfigureEvents(this IServiceProvider provider)
         {
             var dispatcher = provider.GetRequiredService<IDispatcher>();
-            if(dispatcher is Dispatcher unboxed) {
+            if (dispatcher is Dispatcher unboxed)
+            {
                 return unboxed;
             }
             throw new Exception("Coravel.ConfigureEvents has been mis-configured.");
         }
+
+
+        /// <summary>
+        /// Registers all Coravel events and listers from the assembly containing the type <typeparamref name="T"/>.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddEventsFromAssembly<T>(this IServiceCollection services)
+        {
+            return services.AddEventsFromAssembly(typeof(T).Assembly);
+        }
+
+        /// <summary>
+        /// Registers all Coravel events and listeners from the given assembly.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="assembly"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddEventsFromAssembly(this IServiceCollection services, Assembly assembly)
+        {
+            var assemblyScanner = new AssemblyScanner(services);
+            services.AddSingleton(assemblyScanner);
+
+            assemblyScanner.AddEventsFromAssembly(assembly);
+
+            return services;
+        }
+
+        /// <summary>
+        /// Registers all Coravel events and listeners from the given assemblies.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="assemblies"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddEventsFromAssemblies(this IServiceCollection services, params Assembly[] assemblies)
```

Comment:
I think it would be much simpler to take the logic you had placed in the `AssemblyScanner` class and inline it as one of the these service methods.

We should expose one method that will scan all loaded assemblies and auto-register them all at once. That way the client doesn't need to manually specify what assemblies they want to use. We'll just scan them all.

Coravel Pro does something like this to auto-register all invocables:

```csharp
            GetAllTypesOfType<IInvocable>()
                .Distinct()
                .ToList()
                .ForEach(i => services.AddTransient(i));
```

If you have any thoughts or I missed something please let me know 👍

Reply:
That makes sense, I'll make the changes in the coming days. I was a bit too locked in on the proposed implementation (with two different calls) that I didn't stop to consider an easier way 🙂

Reply:
It's all good. Thanks! No rush 😊